---
title: "04 Gene Ontology"
author: "EL Strand"
output:
  github_document: default
  pdf_document:
    keep_tex: yes
  html_document:
    toc: yes
    toc_depth: 6
    toc_float: yes
editor_options: 
  chunk_output_type: inline
---

# Finding gene ontology terms 

## Load libraries 

```{r}
## GO SEQ 
#BiocManager::install("GO.db")
library(GO.db)
#BiocManager::install("goseq")
library(goseq)
library(ontologyIndex)

## GENERAL
library(ggplot2)
library(tidyverse)
```

## Load gene annotation information data 

```{r}
eggNOGG <- read.delim2("data/Pocillopora_acuta_HIv2.genes.EggNog_results.txt") %>% dplyr::rename(gene = X.query)
length(unique(eggNOGG$gene)) #16,784

gff <- read.delim2("data/Pocillopora_acuta_HIv2.genes.gff3", header=F,
                   col.names = c("Scaffold", "Program", "Region", "Start", "Stop", "V6", "V7", "V8", "gene")) %>%
  dplyr::select(gene, Start, Stop) %>% 
  filter(grepl("ID=", gene)) %>%
  mutate(Length=Stop-Start,
         gene = gsub("ID=", "", gene)) %>%
  dplyr::select(gene, Length)
length(unique(gff$gene)) #33,730

gene_meta <- left_join(eggNOGG, gff, by = "gene") #16,784

go_slim <- get_ontology("http://current.geneontology.org/ontology/subsets/goslim_generic.obo")
names(go_slim)

go_slim_df <- tibble(
  GO_ID = names(go_slim$name),
  name = unname(go_slim$name)
)
head(go_slim_df)
```

## Load data (Input)

The core of GO enrichment analysis involves comparing the input gene set against the background of all genes in the genome. The analysis aims to identify GO terms that are overrepresented in the input set compared to what would be expected by chance. 

```{r}
## Input set 
hypolist_D <- read.csv("data/WGBS/DMG_hypometh_diploidy.csv") %>% dplyr::select(gene) %>% left_join(., gene_meta)
hypolist_T1 <- read.csv("data/WGBS/DMG_hypometh_triploidy1.csv") %>% dplyr::select(gene) %>% left_join(., gene_meta)
hypolist_T2 <- read.csv("data/WGBS/DMG_hypometh_triploidy2.csv") %>% dplyr::select(gene) %>% left_join(., gene_meta)
hyperlist_D <- read.csv("data/WGBS/DMG_hypermeth_diploidy.csv") %>% dplyr::select(gene) %>% left_join(., gene_meta)
hyperlist_T1 <- read.csv("data/WGBS/DMG_hypermeth_triploidy1.csv") %>% dplyr::select(gene) %>% left_join(., gene_meta)
hyperlist_T2 <- read.csv("data/WGBS/DMG_hypermeth_triploidy2.csv") %>% dplyr::select(gene) %>% left_join(., gene_meta)
```

## Load background data 

```{r}
load("data/WGBS/meth_table5x_filtered2.RData") #1,799 genes 
background_df <- as.data.frame(unique(meth_table5x_filtered2$gene)) %>% dplyr::rename(gene = 1) %>%
  left_join(., gene_meta, by = "gene")

background_genes <- c(t(background_df$gene))
background_genes_ID <- background_df$gene
background_genes_lengths <- background_df$Length
```

## Creating GO ID file based on eggNOGG txt file annotations 

```{r}
GO_terms <- gene_meta %>% dplyr::select(gene, GO_ID = GOs) %>%
  mutate(across(everything(), gsub, pattern = ",", replacement = ";"),
         across(everything(), gsub, pattern = "-", replacement = NA)) %>%
  separate_rows(GO_ID, sep = ";")

## move hyper and hypo files to mac book and run on that script then move output back here 
```

## Writing goenrichment function 

PWF = Probability Weighting Function

```{r}
goenrich <- function(filename, identifier){
  
  # Identify list of genes from filename (input)
  genes_of_interest <- as.character(hypolist_D$gene)
  
  # construct vector with 1 for DMG and 0 for others 
  gene_vector = as.integer(background_genes %in% genes_of_interest)
  names(gene_vector)=background_genes # set names 
  
  # Weight vector by length of gene 
  ## col names = row name, DMgenes, bias.data, pwf
  DMG.pwf <- nullp(gene_vector, background_genes_ID, bias.data = background_genes_lengths)
  
  # Find GO enriched terms 
  ## col names of GO = category, overrep p value, underrep p value, numDEinCat, numinCat
  GO <- goseq(
    DMG.pwf, background_genes_ID, gene2cat = GO_terms,
    test.cats = c("GO:CC", "GO:BP", "GO:MF"),
    method = "Wallenius",
    use_genes_without_cat = TRUE
  )
  
  GO <- GO[order(GO$over_represented_pvalue),]
  colnames(GO)[1] <- "GO_ID"
  
  # Adding GO Slim categories 
  Slim <- merge(GO, go_slim_df, by = "GO_ID")
  Slim <- Slim[!duplicated(Slim$GO_ID),]
  
  ## Filtering for p<0.05
  filename_sigGO <- Slim %>%
    filter(over_represented_pvalue < 0.05) %>%
    arrage(., ontology, term, over_represented_pvalue)
  
  ## Formatting GO columns 
  filename_GO <- filename %>%
    dplyr::rename(GO_ID = GOs) %>% 
    mutate(across(everything(), gsub, pattern = ",", replacement = ";"),
         across(everything(), gsub, pattern = "-", replacement = NA)) %>%
    separate_rows(GO_ID, sep = ";")
  
  ## Merge sig meth genes with goslim
  filename_GOslim <- left_join(filename_GO, filename_sigGO, by = "GO_ID")
  
  return(filename_GOslim)
  
  }
```

Running that on the 6 gene lists 

```{r}
GOenrich_hypolist_D <- goenrich(hypolist_D, hypo_diploidy)
```
















