---
title: "Targeted gene expression"
author: "EL Strand"
output:
  github_document: default
  pdf_document:
    keep_tex: yes
  html_document:
    toc: yes
    toc_depth: 6
    toc_float: yes
editor_options: 
  chunk_output_type: inline
---

## Load libraries 

```{r}
library(plyr)
library(dplyr)
library(ggplot2)
library("RColorBrewer")
library("tidyverse")
library("ggpubr")
library("DESeq2")
library("genefilter")
library(Rmisc)
library(lme4)
library(car)
library(purrr)
library(lubridate)
library(Hmisc)
library(naniar)
library(stats)
library(forcats)
library(data.table)
library(writexl)
library(ggpmisc)
library(multcomp)
library("genefilter")
library("DESeq2")
library("RColorBrewer")
library("WGCNA")
library("flashClust")
library("gridExtra")
library("ComplexHeatmap")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("reshape2")
library(readxl)
```

## Load data 

```{r}
GEdata <- read.delim("data/gene expression/Pocillopora_acuta_KBHIv2.gentrome.fa.gz.salmon.numreads.matrix") %>%
  dplyr::rename(gene_id = Name) %>% gather(., "Sample.ID", "count", 2:120) %>%
  mutate(Sample.ID = str_remove(Sample.ID, ".*_")) 

meta <- read.csv("data/metadata/meth_pattern_groups.csv") %>% dplyr::select(-X) %>%
  mutate(Sample.ID = as.character(Sample.ID)) 

GEdata_filtered <- GEdata %>% filter(Sample.ID %in% meta$Sample.ID) %>%
  spread(., Sample.ID, count) %>%
  mutate(gene_id = str_replace_all(gene_id, "KB", ""))
```

## Data filtering: PoverA and genefilter

Conduct data filtering, this includes:  

*pOverA*: Specifying the minimum count for a proportion of samples for each gene. Filter in the package "genefilter". Pre-filtering our dataset to reduce the memory size dataframe, increase the speed of the transformation and testing functions, and improve quality of statistical analysis by removing low-coverage counts. Removed counts could represent outliers in the data and removing these improves sensitivity of statistical tests. 

### Pacuta 

**Methylation pattern group**: Here, we are using a pOverA of 0.294. This is because we have 51 samples with a minimum of n=15 samples per group. Therefore, we will accept genes that are present in 15/51 = 0.294 of the samples because we expect different expression by ploidy. We are further setting the minimum percentage of genes to 1, such that 29.4% of the samples must have a counts minimum of 10.  

``` {r, echo=TRUE, warning=FALSE, message=FALSE}
Pacuta_matrix <- GEdata_filtered %>% column_to_rownames(., var = "gene_id")
Pfilt <- filterfun(pOverA(0.294,10))

#create filter for the counts data
Pgfilt <- genefilter(Pacuta_matrix, Pfilt)

#identify genes to keep by count filter
Pkeep <- Pacuta_matrix[Pgfilt,]

#identify gene lists
Pn.keep <- rownames(Pkeep)

#gene count data filtered in PoverA, P percent of the samples have counts over A
Pacuta_data_filt <- as.data.frame(Pacuta_matrix[which(rownames(Pacuta_matrix) %in% Pn.keep),])

#How many rows do we have before and after filtering?
nrow(Pacuta_matrix) #Before = 33,259
nrow(Pacuta_data_filt) #After = 19,128 for ploidy 
#Filtering removed 14,131 in ploidy analysis 
```

In order for the DESeq2 algorithms to work, the SampleIDs on the treatmentinfo file and count matrices have to match exactly and in the same order. The following R clump will check to make sure that these match.

```{r}
#Checking that all row and column names match. Should return "TRUE"
all(meta$Sample.ID %in% colnames(Pacuta_data_filt))
all(meta$Sample.ID == colnames(Pacuta_data_filt)) 
```
## Read normalization
We are now going normalize our read counts using VST-normalization in DESeq2

### Construct the DESeq2 dataset

Create a DESeqDataSet design from gene count matrix and labels. Here we set the design to look at time_point to test for any differences in gene expression across timepoints.

```{r}
Pacuta_gdds_dev <- DESeqDataSetFromMatrix(countData = round(Pacuta_data_filt),
                              colData = meta,
                              design = ~meth_exp_group)

## combine DESeq matrix so can directly compare and normalize them together 
pacuta_forall <- Pacuta_data_filt %>% 
  rownames_to_column(., var="gene_id") %>%
  gather(., "Sample.ID", "count", 2:52) #120 for all pacuta 

all_data_filt <- pacuta_forall %>% 
  spread(., Sample.ID, count) %>%
  column_to_rownames(., var="gene_id")
```

#### Log-transform the count data

First we are going to log-transform the data using a variance stabilizing transforamtion (VST). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects.

To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4. Otherwise, there could be artefacts in our results.

```{r}
Pacuta_SF.gdds_dev <- estimateSizeFactors(Pacuta_gdds_dev) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than four to use vst
print(sizeFactors(Pacuta_SF.gdds_dev)) #View size factors
```

Our size factors are all less than 4, so we can use VST!

```{r}
Pacuta_gvst_dev <- vst(Pacuta_gdds_dev, blind=FALSE) #apply a variance stabilizing transforamtion to minimize effects of small counts and normalize wrt library size
head(assay(Pacuta_gvst_dev), 3) #view transformed gene count data

## create data matrix 
pacuta_counts <- as.data.frame(assay(Pacuta_gvst_dev)) 
```

## Gene expression counts and CV versus DNA methylation levels

Read in methylation data and transform counts, methylation level, and expression CV. 

This uses the mean but I could switch this to the median. The mean is more representative after removing the loci with 0 % methylation.

```{r}
### Methylation data 
gene.stats.DMGs <- read.csv("data/WGBS/DMG_statistics.csv") %>% 
  dplyr::select(Sample.ID, gene, median.gene, mean.gene) %>% distinct() %>%
  mutate(Sample.ID = as.character(Sample.ID)) 

### Gene expression data 
GE.stats.formeth <- pacuta_counts %>% 
  rownames_to_column(., var="gene") %>%
  gather("Sample.ID", "count", 2:52) %>% filter(!is.na(count))  

EXP_METH_compare <- full_join(gene.stats.DMGs, GE.stats.formeth, by = c("Sample.ID", "gene")) %>%
  left_join(., meta, by = "Sample.ID") %>% na.omit() %>% gather("measurement", "value", 3:5)

comparison_summary <- summarySE(EXP_METH_compare, measurevar = c("value"), 
                               groupvars = c("measurement", "gene", "meth_exp_group"))

comparison_summary_GE <- comparison_summary %>% filter(measurement == "count") %>%
  dplyr::select(gene, meth_exp_group, value, sd) %>% dplyr::rename(count = value) %>% dplyr::rename(count_sd = sd)

comparison_summary_meth <- comparison_summary %>% filter(measurement == "mean.gene") %>% 
  dplyr::select(gene, meth_exp_group, value, sd) %>% dplyr::rename(per.meth = value) %>% dplyr::rename(meth_sd = sd) 

comparison_summary2 <- full_join(comparison_summary_GE, comparison_summary_meth, by = c("gene", "meth_exp_group")) %>%
  mutate(CV = (count_sd/count),
         inv_CV = (1/CV),
         log_expression = log10(count+1)) %>%
  filter_all(all_vars(!is.infinite(.)))
```

Methylation vs. Gene expression

```{r}
### statistic
ANOVA <- aov(count~meth_exp_group*per.meth, data = comparison_summary2)
summary(ANOVA)

### figure with stats
comparison_summary2 %>%
  ggplot(., aes(x=per.meth, y=count, color=meth_exp_group)) + 
  geom_point(alpha=0.2, size=0.75) + 

  labs(
    y = "Gene expression",
    x = "Mean gene methylation (%)"
  )+
  
  stat_poly_line(linewidth=1) +
  scale_colour_manual(values = c("skyblue3", "olivedrab4", "darkgreen")) +
  stat_poly_eq(vjust=0.1, hjust=0.2) +
  
  # Add y-axis buffer of 0.2
  scale_y_continuous(limits = c(min(comparison_summary2$count) - 0.05,
                                 max(comparison_summary2$count) + 0.14)) +
  
  ggtitle(
    subtitle = paste0("ANOVA Methylation p<0.0001"),
    label = paste0("")
  ) +
  
  theme_bw() +
  theme(legend.position = "none",
        axis.text.y = element_text(size=10, color="black"),
        axis.text.x = element_text(size=10, color="black"),
        plot.title = element_text(size=10, color="grey55", face = "italic"),
        plot.subtitle = element_text(size=10, color="grey55", face = "italic"),
        axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size=11, face="bold"),
        axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size=11, face="bold")
  )

ggsave("data/figures/Fig4A_Methylation_GE.png", width = 5.5, height = 4.5)
```

Methylation vs. Gene expression CV 

```{r}
## CV or inv_CV 

### statistic
comparison_summary2$meth_exp_group <- as.factor(comparison_summary2$meth_exp_group)
ANOVA <- aov(inv_CV~meth_exp_group*per.meth, data = comparison_summary2)
summary(ANOVA)
summary(glht(ANOVA, linfct = mcp(meth_exp_group = "Tukey")))

lm_model <- lm(inv_CV ~ meth_exp_group*per.meth, data = comparison_summary2)
summary(lm_model)

### figure with stats
comparison_summary2 %>%
  ggplot(., aes(x=per.meth, y=inv_CV, color=meth_exp_group)) + 
  geom_point(alpha=0.2, size=0.75) + 

  labs(
    y = "Gene expression (CV-1)",
    x = "Mean gene methylation (%)"
  )+
  
  stat_poly_line(linewidth=1) +
  scale_colour_manual(values = c("skyblue3", "olivedrab4", "darkgreen")) +
  stat_poly_eq(vjust=0.1, hjust=0.2) +
  
  # Add y-axis buffer of 0.2
  # scale_y_continuous(limits = c(min(comparison_summary2$CV) - min(comparison_summary2$CV)*0.01,
  #                                max(comparison_summary2$CV) + max(comparison_summary2$CV)*0.02)) +
  
  ggtitle(
    label = paste0("ANOVA Methylation p<0.0001; Ploidy p<0.0001"),
    subtitle = paste0("TUKEY POST-HOC T2-D p=0.0017; T2-T1 p<0.0001")
  ) +
  
  theme_bw() +
  theme(legend.position = "none",
        axis.text.y = element_text(size=10, color="black"),
        axis.text.x = element_text(size=10, color="black"),
        plot.title = element_text(size=10, color="grey55", face = "italic"),
        plot.subtitle = element_text(size=10, color="grey55", face = "italic"),
        axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0), size=11, face="bold"),
        axis.title.x = element_text(margin = margin(t = 10, r = 0, b = 0, l = 0), size=11, face="bold")
  )

ggsave("data/figures/Fig4A_Methylation_CV.png", width = 5.5, height = 4.5)
```


## Targeted Epigenetic machinery gene expression 

Determining which epigenetic machinery genes have significantly different expression 

```{r}
ploidy_DEG_wfunction <- read.csv(file="data/gene expression/3-Pacu-group-DEG-GOenrichment-results.csv") 

eggNOGG <- read.delim2("data/gene expression/Pocillopora_acuta_HIv2.genes.EggNog_results.txt") %>%
  dplyr::select(gene = X.query, Description) %>%
  filter(grepl("acetylase|methylase|methyltransferase|PHD|WD40|MBD|UHRF|Methylcytosine|S-adenosylmethionine|methionine synthase", 
               Description, ignore.case = TRUE)) 

## 208 genes matching epigenetic machinery names
length(unique(eggNOGG$gene))
eggNOGG %>% write_xlsx("data/metadata/Epigenetic_machinery_genes.xlsx")

epigenetic_machinery_genes <- left_join(eggNOGG, pacuta_counts %>% rownames_to_column(var="gene")) %>%
  gather("Sample.ID", "count", 3:last_col()) %>% left_join(., meta, by = "Sample.ID") %>%
  group_by(Sample.ID) %>% filter(!is.na(count))

## 191 / 208 genes we have data for 
length(unique(epigenetic_machinery_genes$gene))

### running ANOVA loop on all genes in this df to see which ones have significant expression differences
epigene_results <- data.frame()

for (i in epigenetic_machinery_genes$gene){
  stats_subset <- epigenetic_machinery_genes %>% subset(gene==i)
  aov <- aov(count ~ meth_exp_group, data=stats_subset) 
  ANOVA <- Anova(aov, type = "III")
  TUKEY <- TukeyHSD(aov)
  
  pvalue <- as.data.table(ANOVA)[2,4]
  pvalue <- pvalue %>% dplyr::rename(pvalue = `Pr(>F)`)
  
  Tukeydf <- as.data.frame(TUKEY$meth_exp_group[,4]) %>% rownames_to_column(., "comparison") %>%
              dplyr::rename(pvalue = `TUKEY$meth_exp_group[, 4]`) %>% spread(comparison, pvalue)
  
  DF <- data.frame(stats_subset[1,1], pvalue, Tukeydf)  
  
  epigene_results <- rbind(epigene_results, DF)
  }

epigene_results_filtered <- epigene_results %>% distinct() %>%
  mutate_if(is.numeric, round, 8) %>% 
  mutate(sig = ifelse(pvalue < 0.050, "significant", "NS"))

## genes with significantly different expression = 88 genes 
significant_genes = epigene_results_filtered %>% filter(pvalue < 0.050) %>% dplyr::select(gene)
```

```{r}
distance_matrix_df <- significant_genes %>% 
  left_join(., pacuta_counts %>% rownames_to_column(var="gene"), by = "gene") %>% 
  column_to_rownames(var = "gene")

# Assuming 'data' is your normalized gene expression matrix
dist_matrix <- dist((distance_matrix_df), method = "euclidean")  # Compute distance matrix
hclust_result <- hclust(dist_matrix, method = "complete")
#hc <- hclust(dist_matrix)
clusters <- cutree(hclust_result, k = 4)

# Create a heatmap with hierarchical clustering results
annotation_col <- meta %>% column_to_rownames(var = "Sample.ID") # Replace 'Group' with your actual metadata column
group_order<-c("D", "T1", "T2")

ordered_samples <- annotation_col[order(factor(annotation_col$meth_exp_group, levels = group_order)), , drop = FALSE]
ordered_indices <- match(rownames(ordered_samples), colnames(distance_matrix_df))
ordered_distance_matrix_df <- (distance_matrix_df[, ordered_indices])

# Create a color palette for annotations (optional)
ann_colors <- list(meth_exp_group = c(D = "skyblue3", T1 = "olivedrab4", T2 = "darkgreen")) 

# Generate heatmap with ordered columns
pdf("data/figures/Figure5_Heatmap_epigenes.pdf")
heatmap_output <- pheatmap(ordered_distance_matrix_df,
         scale = "row",                      
         clustering_distance_rows = "euclidean",  
         clustering_method = "complete",     
         color = colorRampPalette(c("navy", "white", "firebrick"))(50),  
         show_rownames = FALSE,               
         show_colnames = FALSE, 
         cluster_cols = FALSE,
         annotation_col = ordered_samples,
         annotation_colors = ann_colors,
         annotation_names_col = FALSE,
         gaps_col = c(15, 35),
         cutree_rows = 4)
dev.off()

row_order <- heatmap_output$tree_row$order

#Extract the cluster assignments:
cluster_assignments <- cutree(heatmap_output$tree_row, k = 4)
ordered_clusters <- cluster_assignments[row_order]
cluster_mapping <- setNames(seq_along(unique(ordered_clusters)), unique(ordered_clusters))

cluster_df <- data.frame(gene = names(cluster_assignments), 
                         cluster = cluster_mapping[as.character(cluster_assignments)]) %>%
  left_join(., eggNOGG %>% dplyr::select(gene, Description), by = "gene") %>%
  left_join(., ploidy_DEG_wfunction %>%
              separate_rows(genes, sep = ",") %>% dplyr::rename(gene = genes) %>%
              mutate(gene = gsub("KB", "", gene)),
            by = "gene"
            )

cluster_df %>% group_by(cluster) %>%
  reframe(counts = n_distinct(gene))

#cluster_df %>% subset(cluster == 1) %>% left_join(., epigene_results_summary, by = "gene")

cluster_df %>% write.csv("data/gene expression/Epigenetic_machinery_genes_heatmapclusters.csv", row.names = FALSE)
cluster_df_annotated <- read_excel("data/gene expression/Epigenetic_machinery_genes_heatmapclusters.xlsx")
```

Trying heatmap from other script to confirm distance is up/down 

```{r}
# All_data <- significant_genes %>% 
#   left_join(., pacuta_counts %>% rownames_to_column(var="gene"), by = "gene") %>% 
#   column_to_rownames(var = "gene")
# All_mat <- as.matrix(All_data) #create matrix and remove gene name 
# group_order<-c("D", "T1", "T2")
# 
# par("mar")
# par(mar=c(1,1,1,1))
# 
# col_labels_df <- All_data %>% rownames_to_column(var = "gene") %>% gather("Sample.ID", "value", 2:last_col()) %>% 
#   dplyr::select(Sample.ID) %>% distinct() %>% left_join(., meta, by = "Sample.ID") %>%
#   mutate(color = case_when(
#     meth_exp_group == "D" ~ "skyblue3",
#     meth_exp_group == "T1" ~ "olivedrab4",
#     meth_exp_group == "T2" ~ "darkgreen"
#   ))
# col_labels <- col_labels_df$color
# 
# pdf("data/figures/Figure5_Heatmap_epigenes_v2.pdf")
# heatmap_df<-
# heatmap.2(as.matrix(All_mat),
#           cexCol = 1.5, 
#           distfun = function(x) as.dist(1 - cor(t(x), use = "pa")), 
#           hclustfun = function(x) hclust(x,method = 'average'),
#           key.xtickfun = function() {
#   breaks = pretty(parent.frame()$breaks)
#   breaks = breaks[c(1,length(breaks))]
#   list(at = parent.frame()$scale01(breaks),labels = breaks)},
#   Colv=TRUE, 
#   
#   col= colorRampPalette(c("navy", "white", "firebrick"))(50), 
#   density.info = "none", 
#   ColSideColors=col_labels,
#   dendrogram = "row",
#   trace = "none", 
#   scale = "row", 
#   labRow = FALSE,
#   labCol = FALSE,
#   sepcolor="none",
#   sepwidth=c(0.1,0.01),
#   lmat = rbind(c(5,4), c(0,1), c(3,2)),
#   keysize=0.1, 
#   key.par = list(mgp = c(1.5, 0.5, 0), cex=0.7), 
#   lhei=c(0.5,0.1,2),
#   lwid = c(1.5,4))
# dev.off()
```


### Boxplot prelim

```{r}
# epigene_results_significant <- significant_genes %>% 
#   left_join(., pacuta_counts %>% rownames_to_column(var="gene"), by = "gene") %>%
#   gather("Sample.ID", "count", 2:last_col()) %>%
#   left_join(., meta, by = "Sample.ID") %>%
#   left_join(., eggNOGG %>% dplyr::select(gene, Description), by = "gene")
# 
# length(unique(epigene_results_significant$gene))
# 
# epigene_results_summary <- summarySE(epigene_results_significant, measurevar = c("count"), 
#                                groupvars = c("gene", "meth_exp_group")) %>% dplyr::select(-N, -sd, -se, -ci) %>%
#   spread(meth_exp_group, count) %>% left_join(., epigene_results_filtered, by = "gene") %>%
#   left_join(., eggNOGG %>% dplyr::select(gene, Description), by = "gene")
# 
# epigene_results_summary %>% write_xlsx("data/Epigenetic_machinery_genes.xlsx")
# 
# epigene_annotated <- read_xlsx("data/Epigenetic_machinery_genes_annotated.xlsx")
# 
# epigene_annotated %>% group_by(Pattern) %>%
#   reframe(n_genes = n_distinct(gene))
# 
# epigene_annotated %>% dplyr::select(-pvalue, -`T1.D`, -`T2.D`, -`T2.T1`, -sig) %>%
#   gather("group", "value", 2:4) %>% 
#   ggplot(., aes(x=group, y=(value))) +
#   labs(
#     y="",
#     x=""
#   ) +
#   geom_line(aes(group = gene), color = 'grey80', lty = "dotted") + 
#   geom_point(width=0.2, aes(fill = group), size=1.5, color = 'black', shape=21, alpha=0.5) +
#   
#   facet_wrap(~Pattern, scales = "free") +
#   scale_fill_manual(values = c("skyblue3", "olivedrab4", "darkgreen")) +
#   theme_bw() +
#   theme(
#     strip.text.x = element_text(size = 8, color = "black", face = "bold"),
#         strip.text.y = element_text(size = 8, color = "black", face = "bold"),
#         strip.background = element_rect(color="black", fill="white", size=0.5, linetype="solid"),
#     legend.position = "none"
#   )
# 
# ggsave("data/figures/Fig5_targetedGE_high.png", width = 5, height = 4)
```


## CO SEQ 

Using co-seq to find expressin pattern modules 

```{r}
# # if (!require("BiocManager", quietly = TRUE))
# #     install.packages("BiocManager")
# # 
# # BiocManager::install("coseq")
# library(coseq)
# library(Biobase)
# library(corrplot)
```

Counts library should be:  
- A standard coseq analysis takes the following form, where counts represents a matrix or data.frame of gene-level counts arising from an RNA-seq experiment (of dimension n x d for n genes and d samples).  
- The "K" parameter represents the number of clusters (or centroids) to be used in the k-means algorithm  
- This means the coseq function will run k-means clustering multiple times, with the number of clusters (K) varying from 2 to 25 (K=2:25)

```{r}
# ### Example dataset
# data("fietz")
# counts <- exprs(fietz)
# conds <- pData(fietz)$tissue
# 
# 
# ## round(Pacuta_data_filt)
# 
# ### METHOD 1: EPIGENE COUNT DF
# ## the example uses raw counts but I'm using vst transformed here.. Is that a problem?
# epigene_counts <- significant_genes %>% 
#   left_join(., pacuta_counts %>% rownames_to_column(var="gene"), by = "gene") %>% 
#   column_to_rownames(var = "gene")
# 
# epigene_conds <- meta$meth_exp_group
# 
# ### METHOD 2: FILTERED DDS MATRIX THEN DESEQQ
# # Assuming epigene_counts is a matrix or data frame with gene names as row names
# genes_to_keep <- rownames(epigene_counts)
# 
# # Filter the Pacuta_gdds_dev object to keep only the genes in epigene_counts
# dds_filtered <- Pacuta_gdds_dev[rownames(Pacuta_gdds_dev) %in% genes_to_keep, ]
# 
# # Run DESeq on the filtered dataset
# dds_filtered <- DESeq(dds_filtered, test="LRT", reduced = ~1)
# res <- results(dds_filtered)
# summary(res)
# 
# ### METHOD #3: raw count with transformation
# epigene_rawcount <- round(Pacuta_data_filt)
# epigene_rawcount <- epigene_rawcount[rownames(epigene_rawcount) %in% genes_to_keep, ]
# 
# ### METHOD #4: Summary of groups so means 
# epigene_summary_coseq <- epigene_results_summary %>% dplyr::select(gene, D, T1, T2) %>%
#   column_to_rownames(var = "gene")
# 
# ## running without the normalization methods provided in the function (b/c already vst normalized)
# rawraw <- coseq(epigene_rawcount, K=2:15, transformation="logclr",norm="TMM", 
#                 model="kmeans", iter.max=70)
#   
# runVST <- coseq(epigene_counts, K=2:25, transformation="none", norm="none", 
#                 model="kmeans", iter.max=30)
# 
# runDDS <- coseq(dds_filtered, K=2:15, verbose=FALSE)
# 
# runSummary <- coseq(epigene_summary_coseq, K=2:15, transformation="none", norm="none", 
#                 model="kmeans", iter.max=70)
#   
# summary(runSummary)
# plot(runSummary, graphs="boxplots") #, conds=epigene_conds
# 
# cluster_id <- as.data.frame(clusters(runSummary)) %>% rownames_to_column(var = "gene") %>%
#   dplyr::rename(cluster = 2) %>%
#   left_join(., epigene_results_summary %>% dplyr::select(gene, D, T1, T2)) %>%
#   gather("meth_exp_group", "value", 3:last_col())
# 
# cluster_id %>%
#   ggplot(., aes(x=meth_exp_group, y=value)) + 
#   #geom_jitter(aes(fill=meth_exp_group), color = 'black', alpha=0.5, size=1, shape=21) +
#   geom_line(aes(group = gene)) +
#   facet_wrap(~cluster)
```









